algorithm/cgal/cgal.cc

```diff
@@ -4712,28 +4804,43 @@ int Fuse(Geometry* geometry) {
   geometry->copyPolygonsWithHolesToGeneralPolygonSets();
   geometry->computeBounds();
 
-  for (int target = -1, nth = 0; nth < size; nth++) {
-    if (!geometry->is_mesh(nth) || geometry->is_empty_mesh(nth)) {
-      continue;
-    }
-    if (target == -1) {
-      target = geometry->add(GEOMETRY_MESH);
-      geometry->setMesh(target, new Surface_mesh());
-      geometry->setIdentityTransform(target);
-    }
-    if (geometry->noOverlap3(target, nth)) {
-      geometry->mesh(target).join(geometry->mesh(nth));
-    } else {
-      Surface_mesh cutMeshCopy(geometry->mesh(nth));
-      if (!CGAL::Polygon_mesh_processing::corefine_and_compute_union(
-              geometry->mesh(target), cutMeshCopy, geometry->mesh(target),
-              CGAL::parameters::all_default(), CGAL::parameters::all_default(),
-              CGAL::parameters::all_default())) {
-        return STATUS_ZERO_THICKNESS;
+  {
+    int target = -1;
+    for (int nth = 0; nth < size; nth++) {
+      if (!geometry->is_mesh(nth) || geometry->is_empty_mesh(nth)) {
+        continue;
       }
+      if (target == -1) {
+        target = geometry->add(GEOMETRY_MESH);
+        geometry->setMesh(target, new Surface_mesh());
+        geometry->setIdentityTransform(target);
+      }
+      if (geometry->noOverlap3(target, nth)) {
+        geometry->mesh(target).join(geometry->mesh(nth));
+      } else if (exact) {
+        Surface_mesh cutMeshCopy(geometry->mesh(nth));
+        if (!CGAL::Polygon_mesh_processing::corefine_and_compute_union(
+                geometry->mesh(target), cutMeshCopy, geometry->mesh(target),
+                CGAL::parameters::all_default(),
+                CGAL::parameters::all_default(),
+                CGAL::parameters::all_default())) {
+          return STATUS_ZERO_THICKNESS;
+        }
+      } else {
+        // TODO: Optimize out unnecessary conversions.
```

Comment:
How many unnecessary conversions are happening?

Reply:
Each boolean is converting to and from the manifold representation, so potentially quite a lot.